vendor/github.com/bitrise-io/stepman/cli/activate_util.go

```diff
@@ -0,0 +1,104 @@
+package cli
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"os"
+
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-io/stepman/stepman"
+)
+
+func compressStep(log stepman.Logger, patchFilePath, targetExecutablePathLatest, targetExecutablePath string) error {
+	patchFileExist, err := pathutil.IsPathExists(patchFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to check if %s path exist: %w", patchFilePath, err)
+	}
+	if patchFileExist {
+		if err := os.Remove(patchFilePath); err != nil {
+			return fmt.Errorf("failed to remove existing patch file: %w", err)
+		}
+	}
+
+	patchFromExist, err := pathutil.IsPathExists(targetExecutablePathLatest)
+	if err != nil {
+		return fmt.Errorf("failed to check if %s path exist: %s", targetExecutablePathLatest, err)
+	}
+	if !patchFromExist {
+		return fmt.Errorf("Latest Step version used for patch (%s) not found", targetExecutablePathLatest)
+	}
+
+	if targetExecutablePath == "" {
+		return nil
+	}
+
+	compressCmd := command.New("zstd", "--patch-from="+targetExecutablePathLatest, targetExecutablePath, "-o", patchFilePath)
```

Comment:
Not urgent for the prototyping, but we should not forget that this brings a new implicit dependency. I think `bitrise setup` should at least check that `zstd` is available in `$PATH` when the offline mode is enabled.